memcached: Build error on ppc853x starting with v1.6.35

With the `memcached` package now standalone (see #6471), we can revisit support for older architectures on DSM 5. I began exploring this and found that the [patch](https://github.com/SynoCommunity/spksrc/blob/master/cross/memcached/patches/ppc853x/000-remove-use-of_pthread_setname_np.patch) we previously needed for `ppc853x` is no longer required due to changes introduced in v1.6.24 [commit](https://github.com/memcached/memcached/commit/d9bf02f6331fc0098af8997d2a571fc4525696fc). 

After removing the patch, I confirmed that `memcached` builds successfully for `ppc853x-5.2` up to version 1.6.34 [build log](https://github.com/user-attachments/files/19483844/job-logs.txt).

However, starting with version 1.6.35, the build fails with the following error [build log](https://github.com/user-attachments/files/19483872/job-logs.txt):

```
In file included from jenkins_hash.c:12:
memcached.h:705: error: redefinition of typedef 'io_queue_t'
memcached.h:694: error: previous declaration of 'io_queue_t' was here
...
make[6]: *** [Makefile:1008: memcached-jenkins_hash.o] Error 1
...
```

Upon comparing v1.6.34 and v1.6.35 [diff](https://github.com/memcached/memcached/compare/1.6.34...1.6.35), it appears this issue stems from a refactor of the IO subsystem [related commit](https://github.com/memcached/memcached/commit/ec1fb560fae7a6dbbce8670f4c30121c51f3ae19).

At this point, I'm unsure whether there's a suitable patch or Makefile flag we can use to work around this on `ppc853x`. Should we consider raising this as an upstream issue?

--- a/cross/memcached/patches/ppc853x/000-remove-use-of_pthread_setname_np.patch
+++ b/cross/memcached/patches/ppc853x/000-remove-use-of_pthread_setname_np.patch
@@ -1,24 +1,24 @@
 # libthread of old ppc archs do not implement pthread_setname_np
 # 
---- thread.c.orig	2023-05-12 23:07:43.000000000 +0000
-+++ thread.c	2023-05-20 17:23:40.278542314 +0000
-@@ -635,9 +635,6 @@
+--- thread.c.orig	2025-02-04 21:32:28.000000000 +0000
++++ thread.c	2025-06-25 16:30:19.094177420 +0000
+@@ -678,9 +678,6 @@
  #define THR_NAME_MAXLEN 16
  void thread_setname(pthread_t thread, const char *name) {
  assert(strlen(name) < THR_NAME_MAXLEN);
--#if defined(__linux__)
+-#if defined(__linux__) && defined(HAVE_PTHREAD_SETNAME_NP)
 -pthread_setname_np(thread, name);
 -#endif
  }
  #undef THR_NAME_MAXLEN
  
---- extstore.c.orig	2023-01-11 06:10:10.000000000 +0000
-+++ extstore.c	2023-05-20 17:27:31.413889750 +0000
-@@ -119,9 +119,6 @@
+--- extstore.c.orig	2025-03-19 21:47:32.000000000 +0000
++++ extstore.c	2025-06-25 16:30:02.018402486 +0000
+@@ -118,9 +118,6 @@
  #define THR_NAME_MAXLEN 16
  static void thread_setname(pthread_t thread, const char *name) {
  assert(strlen(name) < THR_NAME_MAXLEN);
--#if defined(__linux__)
+-#if defined(__linux__) && defined(HAVE_PTHREAD_SETNAME_NP)
 -pthread_setname_np(thread, name);
 -#endif
  }

--- a/cross/memcached/patches/ppc853x/001-ajdust-typedefs-for-older-gcc.patch
+++ b/cross/memcached/patches/ppc853x/001-ajdust-typedefs-for-older-gcc.patch
@@ -1,0 +1,14 @@
+# Adjust typedef of io_queue_t to fix build with gcc 4.3.7 (gcc < 4.5)
+# 
+--- memcached.h.orig	2025-03-18 22:16:07.000000000 +0000
++++ memcached.h	2025-06-25 16:50:45.821775153 +0000
+@@ -691,8 +691,7 @@
+ 
+ typedef STAILQ_HEAD(iop_head_s, _io_pending_t) iop_head_t;
+ typedef struct _io_pending_t io_pending_t;
+-typedef struct io_queue_s io_queue_t;
+-typedef void (*io_queue_stack_cb)(io_queue_t *q);
++typedef void (*io_queue_stack_cb)(struct io_queue_s *q);
+ typedef void (*io_queue_cb)(io_pending_t *pending);
+ // IO pending objects are created and stacked into this structure. They are
+ // then sent off to remote threads.

